tasks/main.yml

```diff
@@ -17,26 +12,281 @@
       {{ ansible_facts['distribution'] ~ '_' ~
       ansible_facts['distribution_version'] }}.yml
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __mssql_vars_file is file
+
+- name: Verify that the user accepts EULA variables
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: "{{ __mssql_server_repository }}"
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
+
+- name: Ensure that the Microsoft SQL Server package is installed
+  package:
+    name: "{{ __mssql_server_packages }}"
+    state: present
+
+- name: Check if the errorlog file exists and its location
+  shell: |
+    set -euo pipefail
+    errorlog="$(/opt/mssql/bin/mssql-conf get filelocation errorlogfile 2>&1 \
+    | sed 's/errorlogfile : //')" || :
+    if [ -s "${errorlog}" ]; then
+      echo "${errorlog}"
+    elif [ -s /var/opt/mssql/log/errorlog ]; then
+      echo /var/opt/mssql/log/errorlog
+    else
+      echo ""
+    fi
+  changed_when: false
+  register: __mssql_errorlog
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Determine if MSSQL was previously set up
+  set_fact:
+    __mssql_is_setup: >-
```

Comment:
Sorry, it might be out of scope of this role... What happens if mssql 2016 or 2017 was previously set up?

Reply:
I think what we talked about was adding a parameter like `mssql_allow_upgrade: true|false` - if `false` (the default) - if the role detects any version of mssql is already setup, then the role will skip - if `true` then the role will do an upgrade - we'll have to get some sort of upgrade procedure from the microsoft guys